godtoolsTests/BehaviorTests/App/Features/Account/Data-DomainInterface/GetUserAccountDetailsRepositoryTests.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  GetUserAccountDetailsRepositoryTests.swift
+//  godtoolsTests
+//
+//  Created by Rachael Skeath on 8/1/25.
+//  Copyright © 2025 Cru. All rights reserved.
+//
+
+import Testing
+@testable import godtools
+import Combine
+import Foundation
+
+struct GetUserAccountDetailsRepositoryTests {
+    
+    private static let userId = "test-user-id"
+    private static let userFamilyName = "Smith"
+    private static let userGivenName = "John"
+    private static let userFullName = "John Smith"
+    private static let userCreatedAt = Date()
+    
+    struct TestArgument {
+        let appLanguage: LanguageCodeDomainModel
+        let joinedOnString: String
+    }
+    
+    @Test(
+        """
+        Given: User is logged in and app language is set
+        When: The user navigates to the Activity page
+        Then: Activity page should populate with the user's name and "joined on" date translated in app language.
+        """,
+        arguments: [
+            TestArgument(appLanguage: .english, joinedOnString: "Joined"),
+            TestArgument(appLanguage: .spanish, joinedOnString: "Unirse")
+        ]
+    )
+    @MainActor func testGetUserAccountDetails(argument: TestArgument) async {
+        
+        let getUserAccountDetailsRepository = Self.getUserDetailsRepository()
+        
+        var cancellables: Set<AnyCancellable> = Set()
+        var userAccountDetails: UserAccountDetailsDomainModel?
+        
+        await confirmation(expectedCount: 1) { confirmation in
+            
+            getUserAccountDetailsRepository.getUserAccountDetailsPublisher(appLanguage: argument.appLanguage.rawValue)
+                .sink { (result: UserAccountDetailsDomainModel) in
+                    
+                    confirmation()
+                    userAccountDetails = result
+                }
+                .store(in: &cancellables)
+            
+            let locale = Locale(identifier: argument.appLanguage.rawValue)
+            let createdAtDateString = Self.getDateFormatter(locale: locale).string(from: Self.userCreatedAt)
+            let joinedOnStringExpected = "\(argument.joinedOnString) \(createdAtDateString)"
+            
+            #expect(userAccountDetails?.name == Self.userFullName)
```

Comment:
Hey @rachaelblue the joined on date test looks good to me.

I'm wondering for the name if we should break that into a separate test?  I was looking at the logic in UserAccountDetailsRepository and there is some specific logic for breaking down the name (https://github.com/CruGlobal/godtools-swift/blob/develop/godtools/App/Features/Account/Data-DomainInterface/GetUserAccountDetailsRepository.swift#L67).

Reply:
Actually, maybe the name test is fine as is.  Some of these tests to me are hard to define.

Reply:
Well now that you say that, I'll go back and add some extra name tests.  Didn't hit all of those cases.  I need to remember to check the code coverage for each file!

Reply:
Okay thanks it doesn't hurt!
I was getting boggled thinking about it for a second because I know some of that logic is dependent on the data layer and what it gives us. Where as the use case domain model itself is just a single name field.

Reply:
Yeah I gotta remember to look at the repo logic itself, and not just the fields in the domain model.

Reply:
What makes it more tricky is some of those rules we need defined from the product or business analytics side.  Those are the tests we really want to capture in behavior tests. Typically I get that from Figma if there are detailed comments on app behavior.  Or, sometimes it's just up to us developers.